src/main/java/org/sigmah/server/service/ProjectModelService.java

```diff
@@ -22,484 +22,503 @@
+				LayoutGroup bannerGroup = new LayoutGroup();
+				bannerGroup.setColumn(i);
+				bannerGroup.setRow(j);
+				bannerGroup.setParentLayout(pMBannerLayout);
+				bannerGroups.add(bannerGroup);
+			}
+		}
+
+		pMBannerLayout.setGroups(bannerGroups);
+
+		pM.setProjectDetails(pMDetails);
+		pM.setProjectBanner(pMBanner);
+
+		projectModelDAO.persist(pM, context.getUser());
+		// Add a root phase : one model has minimum one phase
+		PhaseModel defaultRootPhase = new PhaseModel();
+		defaultRootPhase.setName("Default root phase");
+		defaultRootPhase.setParentProjectModel(pM);
+		defaultRootPhase.setDisplayOrder(0);
+		Layout phaseLayout = new Layout();
+		phaseLayout.setColumnsCount(1);
+		phaseLayout.setRowsCount(1);
+
+		LayoutGroup phaseGroup = new LayoutGroup();
+		phaseGroup.setTitle("Default phase group");
+		phaseGroup.setColumn(0);
+		phaseGroup.setRow(0);
+		phaseGroup.setParentLayout(phaseLayout);
+
+		List<LayoutGroup> phaseGroups = new ArrayList<LayoutGroup>();
+		phaseGroups.add(phaseGroup);
+		phaseLayout.setGroups(phaseGroups);
+
+		defaultRootPhase.setLayout(phaseLayout);
+		em().persist(defaultRootPhase);
+
+		LogFrameModel defaultLogFrame = createDefaultLogFrameModel(pM);
+		em().persist(defaultLogFrame);
+
+		pM.setRootPhaseModel(defaultRootPhase);
+		pM.setLogFrameModel(defaultLogFrame);
+
+		return projectModelDAO.persist(pM, context.getUser());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public ProjectModel update(final Integer entityId, final PropertyMap changes, final UserExecutionContext context) {
+
+		LOG.debug("Begins update the model.");
+
+		ProjectModel model = projectModelDAO.findById(entityId);
+
+		if (model == null) {
+			throw new IllegalArgumentException("No ProjetModel can be found with id #" + entityId);
+		}
+
+		if (changes.get(AdminUtil.PROP_PM_NAME) != null) {
+			model.setName((String) changes.get(AdminUtil.PROP_PM_NAME));
+		}
+		if (changes.get(AdminUtil.PROP_PM_STATUS) != null) {
+			ProjectModelStatus newStatus = changes.get(AdminUtil.PROP_PM_STATUS);
+			if (frameworkDAO.countNotImplementedElementsByProjectModelId(model.getId()) > 0) {
+				throw new IllegalArgumentException("A framework requirement was not entirely fulfilled.");
+			}
+
+			model.setStatus(newStatus);
+		}
+		if (changes.get(AdminUtil.PROP_PM_USE) instanceof ProjectModelType) {
+			for (final ProjectModelVisibility v : model.getVisibilities()) {
+				if (context.getUser().getOrganization().getId().equals(v.getOrganization().getId())) {
+					v.setType((ProjectModelType) changes.get(AdminUtil.PROP_PM_USE));
+					em().merge(v);
+				}
+			}
+		}
+		if (changes.containsKey(AdminUtil.PROP_PM_MAINTENANCE_DATE)) {
+			final Object maintenanceDate = changes.get(AdminUtil.PROP_PM_MAINTENANCE_DATE);
+			if(maintenanceDate instanceof Date) {
+				model.setDateMaintenance((Date)maintenanceDate);
+			} else {
+				model.setDateMaintenance(null);
+			}
+		}
+		if (changes.containsKey(AdminUtil.PROP_PM_DEFAULT_TEAM_MEMBER_PROFILES)) {
```

Comment:
Three things:

- `PropertyMap#get` is a generic method that deal with the cast for you. So should declare et `Set<Integer>` instead of an `Object`
- `findByIds` returns an empty collection for an empty id collection
- ` ! defaultProfileIds instanceof Set` should never occur (given the code, set is always called with a Set<Integer>`)

So, it seems that this part (line 277 to 289) could be replaced by simply

```java
if (changes.containsKey(AdminUtil.PROP_PM_DEFAULT_TEAM_MEMBER_PROFILES)) {
	Set<Integer> defaultProfileIds = changes.get(AdminUtil.PROP_PM_DEFAULT_TEAM_MEMBER_PROFILES);
	List<Profile> profiles = profileDAO.findByIds(defaultProfileIds);
	model.setDefaultTeamMemberProfiles(profiles);
}
```